eox_core/api/v1/tests/test_grades.py

```diff
@@ -0,0 +1,150 @@
+        grade_factory.return_value.return_value.read.return_value.percent = 0.5
+        grade_factory.return_value.return_value.read.return_value.subsection_grades = []
+        get_courseware_courses.return_value.get_course_by_id.return_value.grading_policy = (
+            {}
+        )
+        params = {
+            "course_id": "course-v1:org+course+run",
+            "username": "test",
+        }
+        expected_response = {"earned_grade": 0.5}
+
+        response = self.client.get(self.url, params)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data, expected_response)
+
+    @patch("eox_core.api.v1.views.get_courseware_courses")
+    @patch("eox_core.api.v1.views.get_valid_course_key")
+    @patch("eox_core.api.v1.views.get_course_grade_factory")
+    @patch("eox_core.api.v1.views.get_enrollment")
+    @patch("eox_core.api.v1.views.get_edxapp_user")
+    @patch_permissions
+    def test_get_grade_detail_no_policy(  # pylint: disable=too-many-arguments
+            self,
+            _,
+            get_edxapp_user,
+            get_enrollment,
+            grade_factory,
+            get_valid_course_key,
+            get_courseware_courses,
+    ):
+        """Test that the GET method works including section details"""
+
+        get_edxapp_user.return_value.username = "test"
+        get_enrollment.return_value = None, None
+        grade_factory.return_value.return_value.read.return_value.percent = 0.5
+        grade_factory.return_value.return_value.read.return_value.subsection_grades = {}
+        params = {
+            "course_id": "course-v1:org+course+run",
+            "username": "test",
+            "detailed": True,
+        }
+        expected_response = {"earned_grade": 0.5, "section_breakdown": []}
+
+        response = self.client.get(self.url, params)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data, expected_response)
+
+    @patch("eox_core.api.v1.views.get_courseware_courses")
+    @patch("eox_core.api.v1.views.get_valid_course_key")
+    @patch("eox_core.api.v1.views.get_course_grade_factory")
+    @patch("eox_core.api.v1.views.get_enrollment")
+    @patch("eox_core.api.v1.views.get_edxapp_user")
+    @patch_permissions
+    def test_get_grade_detail_policy(  # pylint: disable=too-many-arguments
+            self,
+            _,
+            get_edxapp_user,
+            get_enrollment,
+            grade_factory,
+            get_valid_course_key,
+            get_courseware_courses,
+    ):
+        """Test that the GET method works with section details and grading policy"""
+
+        get_edxapp_user.return_value.username = "test"
+        get_enrollment.return_value = None, None
+        grade_factory.return_value.return_value.read.return_value.percent = 0.5
+        grade_factory.return_value.return_value.read.return_value.subsection_grades = {}
```

Comment:
why don't you add a real subsection_grades instead of {}? It could be useful as an example

Reply:
Mostly because then I would have to mock a subsection grade object and the test becomes a little too big. One thing I could do is make a test for the helper function `_section_breakdown` and check if it does return what is supposed to. This test already checks that `_section_breakdown` returns an empty list when passed an empty dict and the view adds it to the response when using `detailed=true`.

Reply:
You're right. Adding a test to `_section_breakdown` would be nice!